DetectorDescription/DDCMS/src/DDFilteredView.cc

```diff
@@ -572,14 +572,14 @@ std::string_view DDFilteredView::get<string_view>(const string& key) {
 
 template <>
 double DDFilteredView::get<double>(const string& key) {
-  double result(0.0);
-
   currentSpecPar_ = find(key);
   if (currentSpecPar_ != nullptr) {
-    result = getNextValue(key);
+    return getNextValue(key);
+  } else {
+    throw cms::Exception("DetectorDescription")
+        << "Request for a non-existent numeric parameter " << key << " Please, check that it is defined in XML "
+        << "and has an eval attribute set.";
```

Comment:
Wouldn't it be better to `LogProblem` so we could get a list of all the bad parameters in one run? With the exception, we have to fix them one-by-one and then re-run.

Reply:
There is no guarantee that a job will finish if a numeric value is not set correctly. One can do a grep on the XML that is faster than running a cmsRun. Alternatively a new script can be written or the script for the units check can be amended to do this. Just an idea  😀